andela/suyabay
# [ #115621885] write a middleware to prevent regular and premium users from accessing specific channels route

#### What does this PR do?
- Prevent regular and premium users from accessing specific channels route
#### Description of Task to be completed?
- Write a middleware to grant only super admin users into the channels endpoint
- Adjust the `channel Endpoint test` in order to pass travis
#### How should this be manually tested?
- Postman is needed in order to test this feature.
- go to `suyabay.app/api/v1/channels` and post a channel without providing a token as header, `User unauthorized due to empty token` will be displayed. If otherwise a token is provided but an invalid token, `User unauthorized due to invalid token` will be displayed. If a token owned by a premium user is provided, the user should see `User unauthorized due to invalid or expired token`. Finally, if the correct token (a superadmin token) is provided, the user will be able to access the route endpoint to post the channel.
#### Any background context you want to provide?

none
#### What are the relevant pivotal tracker stories?

 115621885
#### Screenshots (if appropriate)

![screen shot 2016-06-08 at 10 58 27 am](https://cloud.githubusercontent.com/assets/14159420/15891223/8018bfea-2d6b-11e6-94f4-960e8984f444.png)

![screen shot 2016-06-08 at 10 58 37 am](https://cloud.githubusercontent.com/assets/14159420/15891237/8a86ff46-2d6b-11e6-8dd4-7bf9923f85e6.png)

![screen shot 2016-06-08 at 10 58 46 am](https://cloud.githubusercontent.com/assets/14159420/15891248/92860a02-2d6b-11e6-99da-80d017416157.png)
#### Questions:

none

[@unicodeveloper](https://github.com/unicodeveloper)
[@andela-oadebayo](https://github.com/andela-oadebayo)
